py-polars/polars/internals/lazy_functions.py

```diff
@@ -1375,12 +1346,30 @@ def pearson_corr(a: str | pli.Expr, b: str | pli.Expr, ddof: int = 1) -> pli.Exp
     │ 0.544705 │
     └──────────┘
 
+    Spearman rank correlation:
+
+    >>> df = pl.DataFrame({"a": [1, 8, 3], "b": [4, 5, 2], "c": ["foo", "bar", "foo"]})
+    >>> df.select(pl.corr("a", "b", method="spearman"))
+    shape: (1, 1)
+    ┌─────┐
+    │ a   │
+    │ --- │
+    │ f64 │
+    ╞═════╡
+    │ 0.5 │
+    └─────┘
+
     """
     if isinstance(a, str):
         a = col(a)
     if isinstance(b, str):
         b = col(b)
-    return pli.wrap_expr(pypearson_corr(a._pyexpr, b._pyexpr, ddof))
+    if method == "pearson":
+        return pli.wrap_expr(pypearson_corr(a._pyexpr, b._pyexpr, ddof))
```

Comment:
Does this method not support `propagate_nans`? That is not documented in the docstring currently, although I guess the best course of action is to support that actually.

Reply:
Added clarification in the docstrings.